### PR TITLE
feat: Add tunnel by default

### PR DESCRIPTION
**Pull request by kylecarbs**
If an access URL is not specified, we will always tunnel.

This is from community-member feedback who exclaimed that it's confusing having the default for `coder server` display a warning message, and I agree.

There is very little (maybe none) in running `coder server` without tunnel and without an access URL, so this seems like overall a much better UX.

### CODE REVIEW

File: cli/server_test.go [hunk 1]
```diff
@@ -127,6 +129,7 @@ func TestServer(t *testing.T) {
 			"server",
 			"--in-memory",
 			"--address", ":0",
+			"--access-url", "example.com",
 			"--access-url", "localhost:3000/",
```
Comment (ammario) [inline cli/server_test.go:133]: Why provide `access-url` twice?

**Review by ammario — APPROVED** [review]

**Comment by ammario** [comment]
(cc @endingwithali this affects docs)

**Comment by endingwithali** [comment]
yes! auto setting tunnel is a great idea. :) 

to 100% clarify - when running coder without an access url, you wont need to edit coder.env?

**Comment by f0ssel** [comment]
Great change 👍

File: install.sh [hunk 2]
```diff
@@ -121,7 +121,7 @@ $1 package has been installed.
 
 To run Coder as a system service:
 
-  # Set up an external access URL or enable CODER_TUNNEL
+  # Set up an external access URL
```
Comment (bpmct) [inline install.sh:124]:
```suggestion
  # Optional) Set up an external access URL
```

File: docs/install/packages.md [hunk 3]
```diff
@@ -7,7 +7,7 @@
 1. Run Coder as a system service.
 
    ```sh
-   # Set up an access URL or enable CODER_TUNNEL
+   # Set up an access URL
```
Comment (bpmct) [inline docs/install/packages.md:10]:
```suggestion
   # Optional) Set up an access URL
```

File: docs/install/docker.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ You can install and run Coder using the official Docker images published on [Git
 
 Docker is required. See the [official installation documentation](https://docs.docker.com/install/).
 
-## Run Coder with built-in database and tunnel (quick)
+## Run Coder with the built-in database (quick)
```
Comment (bpmct) [inline docs/install/docker.md:7]: Let's still mention we're using the tunnel here, both are configured here based on the lack of flags.

**Review by bpmct — APPROVED** [review]

**Comment by ammario** [comment]
I assume this change won't break existing deployments?

**Comment by kylecarbs** [comment]
@ammario it won't _unless_ they aren't specifying an access URL, but even then it won't break them.

**Pull request merged**